lib/cpp/src/thrift/transport/TServerSocket.cpp

```diff
@@ -659,6 +659,10 @@ shared_ptr<TSocket> TServerSocket::createSocket(THRIFT_SOCKET clientSocket) {
   }
 }
 
+boost::shared_ptr<THRIFT_SOCKET> TServerSocket::getInterruptSocket() {
```

Comment:
Perhaps the interrupt socket and flag should be protected so they are accessible to subclasses of TSocket instead of this?  It's more of a style comment however, I wouldn't gate check-in on changing it.